File: wifi4eu-common/src/main/java/wifi4eu/wifi4eu/common/dto/model/BeneficiaryDTO.java
package wifi4eu.wifi4eu.common.dto.model;

import java.io.Serializable;
import java.util.List;

public class BeneficiaryDTO implements Serializable {
    private UserDTO user;
    private List<MayorDTO> mayors;
    private List<MunicipalityDTO> municipalities;
    private boolean representsMultipleMunicipalities;
    private String associationName;
    private int organisationId;
    private String lang;

    public BeneficiaryDTO() {
    }

    public BeneficiaryDTO(UserDTO user, List<MayorDTO> mayors, List<MunicipalityDTO> municipalities, boolean representsMultipleMunicipalities, String associationName, int organisationId, String lang) {
        this.user = user;
        this.mayors = mayors;
        this.municipalities = municipalities;
        this.representsMultipleMunicipalities = representsMultipleMunicipalities;
        this.associationName = associationName;
        this.organisationId = organisationId;
        this.lang = lang;
    }

    public UserDTO getUser() {
        return user;
    }

    public void setUser(UserDTO user) {
        this.user = user;
    }

    public List<MayorDTO> getMayors() {
        return mayors;
    }

    public void setMayors(List<MayorDTO> mayors) {
        this.mayors = mayors;
    }

    public List<MunicipalityDTO> getMunicipalities() {
        return municipalities;
    }

    public void setMunicipalities(List<MunicipalityDTO> municipalities) {
        this.municipalities = municipalities;
    }

    public boolean isRepresentsMultipleMunicipalities() {
        return representsMultipleMunicipalities;
    }

    public void setRepresentsMultipleMunicipalities(boolean representsMultipleMunicipalities) {
        this.representsMultipleMunicipalities = representsMultipleMunicipalities;
    }

    public String getAssociationName() {
        return associationName;
    }

    public void setAssociationName(String associationName) {
        this.associationName = associationName;
    }

    public int getOrganisationId() {
        return organisationId;
    }

    public void setOrganisationId(int organisationId) {
        this.organisationId = organisationId;
    }

    public String getLang() {
        return lang;
    }

    public void setLang(String lang) {
        this.lang = lang;
    }
}

File: wifi4eu-abac/src/main/angular/src/app/shared/swagger/model/BeneficiaryDTO.ts
import * as models from './models';
import {Type} from "class-transformer";

export interface BeneficiaryDTO {
    user?: models.UserDTO;

    mayors?: Array<models.MayorDTO>;

    municipalities?: Array<models.MunicipalityDTO>;

    representsMultipleMunicipalities?: boolean;

    associationName?: string;

    organisationId?: number;

    lang?: string;

}




/*default implementation one might extend from (or use as is) */


export class BeneficiaryDTOBase  implements BeneficiaryDTO{
    
    user?:  models.UserDTOBase ;

    @Type(() => models.MayorDTOBase)
    mayors?:  models.MayorDTOBase[] ;

    @Type(() => models.MunicipalityDTOBase)
    municipalities?:  models.MunicipalityDTOBase[] ;


    representsMultipleMunicipalities?:  boolean ;


    associationName?:  string ;


    organisationId?:  number ;


    lang?:  string ;

}
